core/src/main/java/org/elasticsearch/action/main/MainResponse.java

```diff
@@ -114,4 +117,24 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.endObject();
         return builder;
     }
+
+    private final static ObjectParser<MainResponse, Void> PARSER = new ObjectParser<>(MainResponse.class.getName(),
+            () -> new MainResponse());
+
+    static {
+        PARSER.declareString((response, value) -> response.nodeName = value, new ParseField("name"));
+        PARSER.declareString((response, value) -> response.clusterName = new ClusterName(value), new ParseField("cluster_name"));
+        PARSER.declareString((response, value) -> response.clusterUuid = value, new ParseField("cluster_uuid"));
+        PARSER.declareString((response, value) -> {}, new ParseField("tagline"));
+        PARSER.declareObject((response, value) -> {
+            response.build = new Build((String) value.get("build_hash"), (String) value.get("build_date"),
+                    (boolean) value.get("build_snapshot"));
+            response.version = Version.fromString((String) value.get("number"));
+            response.available = true;
+        }, (parser, context) -> parser.map(), new ParseField("version"));
```

Comment:
does using object parser cause us to be strict with potentially newly added fields to the response? Maybe it is time to investigate what we can do to make our response parsers more understanding (still having a setting that allows to enable a strict mode). I am asking because I remember I added to `cluster_uuid` to this response with 5.0. In general shall we start not expecting fields that may not be there, and  being more flexible with fields that are there although unexpected.